content/articles/creating-custom-expandable-fab/index.md

```diff
@@ -0,0 +1,463 @@
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android"
+
+    android:fillAfter="true">
+    <translate
+        android:duration="300"
+        android:fromYDelta="300%"
+        android:toYDelta="0%" />
+
+    <scale
+        android:pivotY="50%"
+        android:pivotX="50%"
+        android:toXScale="0.9"
+        android:toYScale="0.9"/>
+
+    <alpha
+        android:fromAlpha="0"
+        android:toAlpha="1"
+        android:duration="800"/>
+
+</set>
+```
+
+2. `to_bottom_animation.xml`:
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android"
+
+    android:fillAfter="true">
+    <translate
+        android:duration="300"
+        android:fromYDelta="0%"
+        android:toYDelta="300%" />
+
+    <scale
+        android:pivotY="50%"
+        android:pivotX="50%"
+        android:fromYScale="0.9"
+        android:fromXScale="0.9"
+        android:toXScale="0.9"
+        android:toYScale="0.9"/>
+
+    <alpha
+        android:fromAlpha="1"
+        android:toAlpha="0"
+        android:duration="150"/>
+
+</set>
+```
+
+3. `rotate_close_animation.xml`:
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android"
+
+    android:fillAfter="true">
+    <rotate
+        android:fromDegrees="45"
+        android:pivotX="50%"
+        android:pivotY="50%"
+        android:toDegrees="0"
+        android:duration="300"/>
+
+</set>
+```
+
+4. `rotate_open_animation.xml`:
+```xml
+<?xml version="1.0" encoding="utf-8"?>
+<set xmlns:android="http://schemas.android.com/apk/res/android"
+
+    android:fillAfter="true">
+    <rotate
+        android:fromDegrees="0"
+        android:pivotX="50%"
+        android:pivotY="50%"
+        android:toDegrees="45"
+        android:duration="300"/>
+
+</set>
+```
+
+### Initializing The Animations
+In the `MainActivity.kt` add the following code to globally initialize the animations:
+```kotlin
+private val rotateOpenAnimation: Animation by lazy {AnimationUtils.loadAnimation(this, R.anim.rotate_open_animation)}
+private val rotateCloseAnimation: Animation by lazy {AnimationUtils.loadAnimation(this, R.anim.rotate_close_animation)}
+private val fromBottomAnimation: Animation by lazy {AnimationUtils.loadAnimation(this, R.anim.from_bottom_animation)}
+private val toBottomAnimation: Animation by lazy {AnimationUtils.loadAnimation(this, R.anim.to_bottom_animation)}
+```
+Below is the complete implementation of the `MainActivty.kt` class:
+```kotlin
+package com.example.expandablefabdemo
+
+import android.content.Intent
+import androidx.appcompat.app.AppCompatActivity
+import android.os.Bundle
+import android.view.View.INVISIBLE
+import android.view.View.VISIBLE
+import android.view.animation.Animation
+import android.view.animation.AnimationUtils
+import android.widget.Toast
+import com.example.expandablefabdemo.databinding.ActivityMainBinding
```

Comment:
I don't think it is necessary to include the package name and imports. Android Studio can handle that.